packages/web/src/components/transaction-details-modal/components/TransactionDetailsContent.tsx

```diff
@@ -97,11 +97,13 @@ const UserDetails = ({ userId }: UserDetailsProps) => {
 const dateAndMetadataBlocks = (transactionDetails: TransactionDetails) => {
   switch (transactionDetails.transactionType) {
     case TransactionType.PURCHASE: {
+      // Rare case: metadata will be null if failed to fetch from identity.
+      if (!transactionDetails.metadata) {
+        return
+      }
       return (
         <>
-          <TransactionPurchaseMetadata
-            metadata={transactionDetails.metadata!}
-          />
+          <TransactionPurchaseMetadata metadata={transactionDetails.metadata} />
```

Comment:
Feels like it should still render the TransactionPurchaseMetadata, which should take in `InAppPurchaseMetadata | undefined` and appropriately default to empty/unknown values for everything.

Reply:
Good call.